RELEASE.md

```diff
@@ -0,0 +1,25 @@
+# Release Process
```

Comment:
should we be honest and delete the release script as well?

Reply:
I think not yet. It will help to give context when moving this to the release tool.
I'm hoping it's a sub command of `sg` instead of the current `yarn release ...` one we currently us.